### PR TITLE
Update to require PHP 7.1+

### PR DESCRIPTION
**Pull request by clue**
This changeset updates the project to require PHP 7.1+ and drop legacy PHP < 7.1 and HHVM as discussed in #312. I'm marking this as a BC break for anybody still stuck on very old PHP versions, but there's little chance this will affect any current projects otherwise.

This PR aims to contain the minimal changeset to update the PHP version requirement only. Follow-up PRs will update our APIs to leverage newer language features.

Builds on top of #310 and others
Refs https://github.com/reactphp/cache/pull/58

### CODE REVIEW

**Review by WyriHaximus — APPROVED** [review]

File: tests/SecureServerTest.php [hunk 1]
```diff
@@ -2,25 +2,21 @@
 
 namespace React\Tests\Socket;
 
+use React\EventLoop\LoopInterface;
+use React\Promise\Promise;
+use React\Socket\ConnectionInterface;
+use React\Socket\Connection;
```
Comment (SimonFrings) [inline tests/SecureServerTest.php]: Nit: I think these should be switched

**Review by SimonFrings — CHANGES_REQUESTED** [review]
I went over everything, only found one small nit, but the rest looks good :)

File: tests/SecureServerTest.php (hunk 1 above)
Comment (clue) [reply tests/SecureServerTest.php]: Good catch, updated as suggested! :shipit:

**Review by SimonFrings — APPROVED** [review]
Nice, let's ship it 🚀

**Pull request merged**